File: Games/RockPaperScissors/src/main/java/jokatu/game/games/rockpaperscissors/input/RockPaperScissorsInputDeserializer.java
package jokatu.game.games.rockpaperscissors.input;

import jokatu.game.games.rockpaperscissors.game.RockPaperScissors;
import jokatu.game.input.TypedSingleKeyInputDeserialiser;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class RockPaperScissorsInputDeserializer extends TypedSingleKeyInputDeserialiser<String, RockPaperScissorsInput> {
	@NotNull
	@Override
	public RockPaperScissorsInput deserialiseTypedSingleValue(@NotNull Map<String, Object> json, @NotNull String choice) {
		return new RockPaperScissorsInput(RockPaperScissors.valueOf(choice));
	}

	@NotNull
	@Override
	protected String getKeyName() {
		return "choice";
	}

	@NotNull
	@Override
	protected Class<String> getType() {
		return String.class;
	}
}

File: Games/RockPaperScissors/src/main/kotlin/jokatu/game/games/rockpaperscissors/input/RockPaperScissorsInputDeserializer.kt
package jokatu.game.games.rockpaperscissors.input

import jokatu.game.games.rockpaperscissors.game.RockPaperScissors
import jokatu.game.input.TypedSingleKeyInputDeserialiser
import org.springframework.stereotype.Component

@Component
class RockPaperScissorsInputDeserializer : TypedSingleKeyInputDeserialiser<String, RockPaperScissorsInput>(String::class.java) {

	override val keyName: String
		get() = "choice"

	override fun deserialiseTypedSingleValue(json: Map<String, Any>, choice: String): RockPaperScissorsInput {
		return RockPaperScissorsInput(RockPaperScissors.valueOf(choice))
	}
}
